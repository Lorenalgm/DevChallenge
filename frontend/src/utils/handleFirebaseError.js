export const AuthErrorHandler = {
    generic:
        'Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.',
    'auth/weak-password':
        'Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.',
    'auth/email-already-in-use': 'O email fornecido ja está em uso.',
    'auth/email-already-exists':
        'O e-mail fornecido já está em uso por outro usuário.',
    'auth/internal-error':
        'Erro interno do servidor. Tente novamente mais tarde.',
    'auth/invalid-email': 'Email inválido.',
    'auth/user-not-found':
        'Não há registro de usuário existente correspondente ao identificador fornecido.',
    'auth/wrong-password': 'Senha digitada não esta correta',
    'resource-exhausted':
        'Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.',
};
