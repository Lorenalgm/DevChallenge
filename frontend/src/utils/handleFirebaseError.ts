export const AuthErrorHandler = (option: string) => {
  switch (option) {
    case "generic":
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
    case "auth/weak-password":
      return "Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.";
    case "auth/email-already-in-use":
      return "O email fornecido ja está em uso.";
    case "auth/email-already-exists":
      return "O e-mail fornecido já está em uso por outro usuário.";
    case "auth/internal-error":
      return "Erro interno do servidor. Tente novamente mais tarde.";
    case "auth/invalid-email":
      return "Email inválido.";
    case "auth/user-not-found":
      return "Não há registro de usuário existente correspondente ao identificador fornecido.";
    case "auth/wrong-password":
      return "Senha digitada não esta correta";
    case "resource-exhausted":
      return "Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.";
    default:
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
  }
};
