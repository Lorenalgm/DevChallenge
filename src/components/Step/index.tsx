import { Box, BoxProps, Heading } from '@chakra-ui/react'

export type StepProps = {
  icon: React.ReactElement
  text: string
} & BoxProps

const Step = ({ icon, text }: StepProps) => (
  <Box
    width="48"
    height="48"
    padding="4"
    display="flex"
    flexDirection="column"
    alignItems="center"
    justifyContent="center"
    color="white"
  >
    {icon}
    <Heading mt="5" fontSize={['md', 'xl', '2xl']} color="primary.500">
      {text}
    </Heading>
  </Box>
)

export default Step
