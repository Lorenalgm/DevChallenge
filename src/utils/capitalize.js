export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
