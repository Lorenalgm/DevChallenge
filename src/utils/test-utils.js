import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
