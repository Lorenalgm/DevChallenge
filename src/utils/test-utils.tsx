import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
