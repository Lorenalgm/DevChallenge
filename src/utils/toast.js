import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
