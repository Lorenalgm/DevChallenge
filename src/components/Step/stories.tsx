import { Story, Meta } from '@storybook/react/types-6-0'
import { FaSearch } from 'react-icons/fa'

import Step, { StepProps } from '.'

export default {
  title: 'Step',
  component: Step
} as Meta

export const Default: Story<StepProps> = (args) => <Step {...args} />

Default.args = {
  icon: <FaSearch size="60" />,
  text: 'Escolha'
}
