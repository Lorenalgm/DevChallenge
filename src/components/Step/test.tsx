import { render, screen } from 'utils/test-utils'

import Step, { StepProps } from '.'

const props: StepProps = {
  text: 'mock text',
  icon: <div data-testid="mock icon" />
}

describe('<Step />', () => {
  it('should render correctly', () => {
    const { container } = render(<Step {...props} />)

    expect(
      screen.getByRole('heading', { name: props.text })
    ).toBeInTheDocument()

    expect(screen.getByTestId('mock icon')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
