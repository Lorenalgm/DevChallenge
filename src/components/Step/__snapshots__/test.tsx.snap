// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should render correctly 1`] = `
.emotion-0 {
  width: var(--chakra-sizes-48);
  height: var(--chakra-sizes-48);
  padding: var(--chakra-space-4);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: var(--chakra-colors-white);
}

.emotion-1 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-md);
  line-height: 1.33;
  margin-top: var(--chakra-space-5);
  color: var(--chakra-colors-primary-500);
}

@media screen and (min-width: 30em) {
  .emotion-1 {
    font-size: var(--chakra-fontSizes-xl);
  }
}

@media screen and (min-width: 48em) {
  .emotion-1 {
    font-size: var(--chakra-fontSizes-2xl);
    line-height: 1.2;
  }
}

<div
  class="emotion-0"
>
  <div
    data-testid="mock icon"
  />
  <h2
    class="chakra-heading emotion-1"
  >
    mock text
  </h2>
</div>
`;
